### DISCUSSION TITLE
[Help]: slidesInView keeps returning undefined

### DISCUSSION THREAD

**Discussion by sepe14** [davidjerleke/embla-carousel#740] [Help] [answered]
### Summary

Hello! I'm trying to get the currently selected slide and to my understanding the docs says this is the way to do it:
``` const [selected, setSelected] = useState(0);
  const [emblaRef, emblaApi] = useEmblaCarousel({ inViewThreshold: 1 });

  const logSlidesInView = useCallback((emblaApi: EmblaCarouselType) => {
    const current = emblaApi.slidesInView();
    console.log(current[0]);
    if (current[0]) {
      setSelected(current[0]);
    }
  }, []);
  useEffect(() => {
    if (emblaApi) emblaApi.on("slidesInView", logSlidesInView);
  }, [emblaApi, logSlidesInView]);```
```
(I'm synchroning some part of the ui with the carousel, that's why I need the useState)

However using the carousel I only got these results:
![kép](https://github.com/davidjerleke/embla-carousel/assets/130492114/527fb9bc-13ca-4931-83a1-813977a92696)

Sorry if it's a trivial question but I couldn't find many examples and discussions about this package. 
Thank you!

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment, marked as answer]
@sepe14 have you seen the [lazy load example](https://www.embla-carousel.com/examples/predefined/#lazy-load) in the docs?

**Comment by sepe14** [comment]
@davidjerleke Thank you very much I've got it working! I should've checked the examples more thoroughly...
